solidity/contracts/liquidity-protection/LiquidityProtectionStats.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/EnumerableSet.sol";
+
+import "./interfaces/ILiquidityProtectionStats.sol";
+import "../utility/Utils.sol";
+import "../utility/SafeMath.sol";
+import "../token/interfaces/IDSToken.sol";
+import "../token/interfaces/IERC20Token.sol";
+
+/**
+ * @dev This contract aggregates the statistics of the liquidity protection mechanism.
+ */
+contract LiquidityProtectionStats is ILiquidityProtectionStats, AccessControl, Utils {
+    using SafeMath for uint256;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    bytes32 public constant ROLE_SUPERVISOR = keccak256("ROLE_SUPERVISOR");
+    bytes32 public constant ROLE_SEEDER = keccak256("ROLE_SEEDER");
+    bytes32 public constant ROLE_OWNER = keccak256("ROLE_OWNER");
+
+    mapping(IDSToken => uint256) public totalPoolAmount;
```

Comment:
Maybe "Amounts"?

Reply:
The function returns a single amount.

Reply:
But it's not a function, it's a state variable that holds multiple amounts...

Reply:
It's also a function. I have no opinion on renaming this, I've used the same name as in the store contract.

Reply:
Never mind.